### PR TITLE
Fix: Empty contact field value set to null on edit

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5117
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This bugfix/feature has been sponsored by @Webmecanik

Editing a contact sets its unset core fields to the empty string ('')
This can be problematic if there are any checks on NULL values

[//]: # ( As applicable: )
#### Steps to reproduce the bug:

1. Create a new contact setting a field like First Name to any value
2. Check the database to see the contact's unset core fields (e.g. firstname, lastname, email) set to NULL
3. Update a lead setting a field like Last Name to any value
4. Check the database to see the contact's unset text core fields (e.g. firstname, lastname, email) set to ''

#### Steps to test this PR:
1.  The same steps like reproduce jsut check if  contact field was set as null in db after edit contact

image

### CODE REVIEW

File: app/bundles/LeadBundle/Entity/CustomFieldEntityTrait.php [hunk 1]
```diff
@@ -119,6 +119,10 @@ public function addUpdatedField($alias, $value, $oldValue = null)
 
         if (property_exists($this, $property) && method_exists($this, $setter)) {
             // Fixed custom field so use the setter but don't get caught in a loop such as a custom field called "notes"
+            // Set empty value as null
+            if (empty($value)) {
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Entity/CustomFieldEntityTrait.php]: This will catch values of 0

Comment (kuzmany) [reply app/bundles/LeadBundle/Entity/CustomFieldEntityTrait.php]: Fixed.
The test should be similar like app\bundles\LeadBundle\Tests\Entity\CustomFieldRepositoryTraitTest.php ?
Have you already some stuff how to write these tests?

**Comment by npracht** [comment]
Tested 👍

**Comment by javjim-mautic** [comment]
Tested and was able to reproduce the bug and the fix

**Pull request merged**